lib/rubocop/cop/rails/pluck_id.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces the use of `ids` over `pluck(:id)` and `pluck(primary_key)`.
+      #
+      # @example
+      #   # bad
+      #   User.pluck(:id)
+      #   user.posts.pluck(:id)
+      #
+      #   def self.user_ids
+      #     pluck(primary_key)
+      #   end
+      #
+      #   # good
+      #   User.ids
+      #   user.posts.ids
+      #
+      #   def self.user_ids
+      #     ids
+      #   end
+      #
+      class PluckId < Cop
+        include RangeHelp
+
+        MSG = 'Use `ids` instead of `%<bad_method>s`.'
+
+        def_node_matcher :pluck_id_call?, <<~PATTERN
+          (send _ :pluck ${(sym :id) (send nil? :primary_key)})
+        PATTERN
+
+        def on_send(node)
+          pluck_id_call?(node) do |argument_node|
+            message = format(MSG, bad_method: build_bad_method(argument_node))
+
+            add_offense(
+              node,
+              location: offense_range(node),
+              message: message
+            )
+          end
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.replace(offense_range(node), 'ids')
+          end
+        end
+
+        private
+
+        def offense_range(node)
+          range_between(node.loc.selector.begin_pos, node.loc.expression.end_pos)
+        end
+
+        def build_bad_method(argument_node)
```

Comment:
I think this method can be removed and user value can be used for argument name.

```diff
diff --git a/lib/rubocop/cop/rails/pluck_id.rb b/lib/rubocop/cop/rails/pluck_id.rb
index 219221098..d07af3a75 100644
--- a/lib/rubocop/cop/rails/pluck_id.rb
+++ b/lib/rubocop/cop/rails/pluck_id.rb
@@ -33,13 +33,10 @@ module RuboCop

         def on_send(node)
           pluck_id_call?(node) do |argument_node|
-            message = format(MSG, bad_method: build_bad_method(argument_node))
+            range = offense_range(node)
+            message = format(MSG, bad_method: range.source)

-            add_offense(
-              node,
-              location: offense_range(node),
-              message: message
-            )
+            add_offense(node, location: range, message: message)
           end
         end

@@ -54,14 +51,6 @@ module RuboCop
         def offense_range(node)
           range_between(node.loc.selector.begin_pos, node.loc.expression.end_pos)
         end
-
-        def build_bad_method(argument_node)
-          if argument_node.sym_type?
-            'pluck(:id)'
-          else
-            'pluck(primary_key)'
-          end
-        end
       end
     end
   end
```

Reply:
Nice 👍